src/serializer/ResidualHeapSerializer.js

```diff
@@ -1163,17 +1173,52 @@ export class ResidualHeapSerializer {
     // TODO #20: add timers
 
     // TODO #21: add event listeners
+
     for (let [moduleId, moduleValue] of this.modules.initializedModules)
       this.requireReturns.set(moduleId, this.serializeValue(moduleValue));
 
+    // Make sure additional functions get serialized.
+    // TODO: make sure those functions get skipped over in all relevant places in serialization/optimization
+    // We have to do this if we're on the first pass so effects get reversed properly
+    let rewrittenAdditionalFunctions: Map<FunctionValue, Array<BabelNodeStatement>> = new Map();
+    let processAdditionalFunctionValues = () => {
+      let additionalFVEffects = this.additionalFunctionValuesAndEffects;
+      if (additionalFVEffects) {
+        for (let [additionalFunctionValue, effects] of additionalFVEffects.entries()) {
+          // Create the new function body
+          let [r, g, ob, pb, co] = effects;
+          let oldGenerator = this.generator;
+          let originalLength = g.length;
+          this.generator = g;
+          this.realm.generator = g;
+          // TODO: should we be ignoring the preludegenerator here?
+          // TODO: does the emitter do the right thing here
+          this.realm.applyEffects([r, new Generator(this.realm), ob, pb, co]);
+          // TODO: make sure property modifications of global object get emitted in correct scope
+          let context = this._getContext();
+          let body = context.serializeGenerator(this.generator);
+          invariant(body.length > 0, body);
+          invariant(additionalFunctionValue instanceof ECMAScriptSourceFunctionValue);
+          rewrittenAdditionalFunctions.set(additionalFunctionValue, body);
+          this.generator = oldGenerator;
+          g.length = originalLength;
```

Comment:
We want the generators being used in the Visitor and the Serializer to be the equal because we pass the generator from the visitor into `_getTarget` to get the corresponding body to emit into. This seemed like the simplest way to do that.